conda_join.py

```diff
@@ -275,39 +383,85 @@ class CondaEnvironmentSpec(NamedTuple):
     pip: list[str]
 
 
-def create_conda_env_specification(
-    requirements: ParsedRequirements,
-) -> CondaEnvironmentSpec:
-    """Create a conda environment specification from `ParsedRequirements`."""
-    conda: list[str | dict[str, str]] = []
-    pip: list[str] = []
-    for dependency, meta in requirements.conda.items():
-        platforms = (
-            extract_matching_platforms(meta.comment) if meta.comment is not None else []
-        )
-        if platforms:
-            unique_platforms = {p.split("-", 1)[0] for p in platforms}
-            dependencies = [
-                {f"sel({_platform})": dependency} for _platform in unique_platforms
-            ]
-            conda.extend(dependencies)
-        else:
-            conda.append(dependency)
+ALLOWED_CONDA = ["unix", "linux", "osx", "win"]
 
-    for dependency, meta in requirements.pip.items():
-        platforms = (
-            extract_matching_platforms(meta.comment) if meta.comment is not None else []
-        )
-        if platforms:
-            for _platform in platforms:
-                selector = _build_pep508_environment_marker([_platform])
-                dep = f"{dependency}; {selector}"
-                pip.append(dep)
+
+def _conda_sel(sel: str) -> str:
+    """Return the allowed `sel(platform)` string."""
+    _platform = sel.split("-", 1)[0]
+    assert _platform in ALLOWED_CONDA, f"Invalid platform: {_platform}"
+    return _platform
+
+
+def _maybe_expand_none(platform_data: dict[Platform | None, dict[str, Meta]]) -> None:
+    if len(platform_data) > 1 and None in platform_data:
+        sources = platform_data.pop(None)
+        for _platform in get_args(Platform):
+            platform_data[_platform] = sources
+
+
+def create_conda_env_specification(  # noqa: PLR0912
+    resolved_requirements: dict[str, dict[Platform | None, dict[str, Meta]]],
+    channels: set[str],
+) -> CondaEnvironmentSpec:
+    """Create a conda environment specification from resolved requirements."""
+    conda_deps = []
+    pip_deps = []
+    for platform_data in resolved_requirements.values():
```

Comment:
First loop and seperate conda and pip

Reply:
Then for conda check for ALLOWED_CONDA by creating dict[CondaPlatform, dict[Platform, list[Meta]] and only add the Meta if the one that exists in the list is not identical. Then in be happy path we should end up with len(meta_list)=1, if not, warn about it and choose one.